Rakefile

```diff
@@ -1,12 +1,28 @@
 require 'rake'
 require 'rspec/core/rake_task'
 require 'puppetlabs_spec_helper/rake_tasks'
+require 'puppet'
+
 begin
   require 'beaker/tasks/test' unless RUBY_PLATFORM =~ /win32/
 rescue LoadError
   #Do nothing, only installed with system_tests group 
 end
 
+
+# If puppet does not support symlinks (e.g., puppet <= 3.5) we cannot use
+# puppetlabs_spec_helper's `rake spec` task because it requires symlink
+# support. Redine `rake spec` without the symlink call and restrict the pattern
+# match to files under the 'unit' directory.
+if !!File::ALT_SEPARATOR and !Puppet::FileSystem.respond_to?(:symlink)
```

Comment:
I'm certainly open to improvements! We could do something like `Puppet.features.microsoft_windows?` ? This was actually taken from the puppetlabs_spec_helper addition @Iristyle pointed me at earlier: https://github.com/puppetlabs/puppetlabs_spec_helper/pull/85/files. Not sure if there was a specific reason we check for `ALT_SEPARATOR` ?